plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/frontend.ts

```diff
@@ -1,9 +1,50 @@
 /**
  * External dependencies
  */
-import { store } from '@wordpress/interactivity';
+import { store, getContext } from '@wordpress/interactivity';
+
+type Attribute = {
+	attribute: string;
+	value: string;
+};
+
+type Context = {
+	variation: Attribute[];
+};
 
 store( 'woocommerce/add-to-cart-with-options', {
```

Comment:
If you change this line to capture the value then you can export a type like:

```
const addToCartWithOptionsStore = store(...
...

export type AddToCartWithOptionsStore = typeof addToCartWithOptionsStore;
```

Then when you call it later such as [here](https://github.com/woocommerce/woocommerce/pull/56378/files#diff-baaa04fb4f93cdad22fe635ee17fdc572b9878c182c6cab5abcd5f33e29f19dbR26)

you can pass it to the store as the generic arg: `store< AddToCartWithOptionsStore >` which will properly type the actions

Reply:
I don't like creating a type by inference since it is not descriptive. But in this case it make sense.

Reply:
As soon as you name the type it becomes descriptive :)